app/src/main/java/com/frinder/frinder/activity/MainActivity.java

```diff
@@ -114,5 +114,17 @@ public void readUserComplete(User user) {
         Log.d(TAG, "Read user from firebase " + user.toString());
         //loggedUser has the user fetched from firebase
         loggedUser = user;
+        LocationUtils locationUtils = new LocationUtils(getBaseContext());
+        locationUtils.startLocationUpdates();
+        locationUtils.getLastLocation(new LocationUtils.LocationUpdate() {
+            @Override
+            public void onSuccess(Location location) {
```

Comment:
We do not call onSuccess in some cases (e.g. permission was not present). In that case, we would not update the location.

Reply:
@donygeorge yes we dont update, onsuccess is only called when location is available
locationClient.getLastLocation()
                .addOnSuccessListener(new OnSuccessListener<Location>() {
                    @Override
                    public void onSuccess(Location location) {
                        // GPS location can be null if GPS is switched off
                        if (location != null) {
                            callback.onSuccess(location);
                        } else {
                            Toast.makeText(context, "Please enable GPS!", Toast.LENGTH_LONG).show();
                        }
                    }
                })
                .addOnFailureListener(new OnFailureListener() {
                    @Override
                    public void onFailure(@NonNull Exception e) {
                        Log.d(TAG, "Error trying to get last GPS location");
                        e.printStackTrace();
                    }
                });